packages/next/client/router.ts

```diff
@@ -133,8 +133,10 @@ export default singletonRouter as SingletonRouter
 // Reexport the withRoute HOC
 export { default as withRouter } from './with-router'
 
-export function useRouter(): NextRouter {
-  return React.useContext(RouterContext)
+export function useRouter<
+  TQueryParams extends ParsedUrlQuery = ParsedUrlQuery
+>(): NextRouter<TQueryParams> {
+  return React.useContext(RouterContext) as NextRouter<TQueryParams>
```

Comment:
Had to typecast because `RouterContext` is not aware of the query params available